### PR TITLE
FSH 3.0 Resolutions - Language Reference and Index

### PR DESCRIPTION
**Pull request by jafeltra**
This PR contains a batch of changes that were agreed up as part of the FSH ballot resolution process. These changes are in both the index page and the language reference page. The following Jira issues were addressed:

[FHIR-42322](https://jira.hl7.org/browse/FHIR-42322) - Rephrase to remove double "but"
[FHIR-42323](https://jira.hl7.org/browse/FHIR-42323) - Clarify which rules Context is preferred over
[FHIR-42324](https://jira.hl7.org/browse/FHIR-42324) - Clarify Extension keywords
[FHIR-42625](https://jira.hl7.org/browse/FHIR-42625) - Overriding the url of an Extension
_Note: In the resolution for this, I added links to the "caret path" section to remain consistent with other places we link to it in the spec and for ease of reference. I also aligned the examples to refer to the same properties that are mentioned in the note, which wasn't specified in the issue, but @cmoesel and I agreed it was helpful._
[FHIR-42319](https://jira.hl7.org/browse/FHIR-42319) - Clarify that alias cannot be used to construct urls
[FHIR-42618](https://jira.hl7.org/browse/FHIR-42618) - Invariant keyword TU statement must be split
_Note: I made the updates suggested in this issue in a second place where they seemed to also apply._
[FHIR-42329](https://jira.hl7.org/browse/FHIR-42329) - Consistent description of keywords
[FHIR-42318](https://jira.hl7.org/browse/FHIR-42318) - Remind reader of relationship between FSH profile and FHIR StructureDefinition, etc.
[FHIR-42316](https://jira.hl7.org/browse/FHIR-42316) - Don't mix profiling best practice with language features
[FHIR-42301](https://jira.hl7.org/browse/FHIR-42301) - Clarify if examples are normative
[FHIR-42321](https://jira.hl7.org/browse/FHIR-42321) - Follow syntax styling guidelines
[FHIR-42311](https://jira.hl7.org/browse/FHIR-42311) - Clarify use of non-ASCII characters
[FHIR-42312](https://jira.hl7.org/browse/FHIR-42312) - Clarify whether $ in alias name removes other name requirements
[FHIR-42313](https://jira.hl7.org/browse/FHIR-42313) - Rearrange item names for readability
_Note: FHIR-42311, FHIR-42312, and FHIR-42313 are all resolved in one commit. Each issue touches on the "item names" section, and I think the resolution in the commit encompasses the intent of each issue, but they were all a little different, so please review to ensure the resolution seems to cover all issues and that the resulting spec is still accurate to our intent for FSH._
_Note (again): The changes in the "item names" section make the restriction on characters in Aliases stronger than SUSHI currently imposes, but we have a Jira issue to track implementing those changes._
[FHIR-42289](https://jira.hl7.org/browse/FHIR-42289) - Remove spreadsheet IG authoring
[FHIR-42288](https://jira.hl7.org/browse/FHIR-42288) - Soften compatibility assertion

At least one reviewer should review each commit alongside its Jira issue to ensure the resolution was followed. Other reviewer(s) should ensure the content makes sense, there are no typos, etc.

**_Note: This PR should be merged using the rebase strategy to maintain commits._**

### CODE REVIEW

**Review by cmoesel — CHANGES_REQUESTED** [review]
These look good. I've made a few suggestions, but they are somewhat subjective -- so feel free to reach out if you'd like to discuss them before making any changes.

File: input/pagecontent/reference.md [hunk 1]
```diff
@@ -979,6 +982,8 @@ Since simple and complex extensions are mutually-exclusive, FSH implementations
 
 Rules types that apply to Extensions are: [Assignment](#assignment-rules), [Binding](#binding-rules), [Cardinality](#cardinality-rules), [Contains (standalone extensions)](#contains-rules-for-extensions), [Contains (inline extensions)](#contains-rules-for-extensions), [Contains (slicing)](#contains-rules-for-slicing), [Flag](#flag-rules), [Insert](#insert-rules), [Obeys](#obeys-rules), [Path](#path-rules), and [Type](#type-rules).
 
+Extension metadata that does not have a dedicated keyword may be specified using assignment rules with [caret paths](#caret-paths) (e.g., `^url`, `^status`, `^purpose`).
+
```
[the remark this answers is not in the document]
Comment (jafeltra) [reply input/pagecontent/reference.md:988]: I think I agree - it feels like it feels more consistent up with the keywords. The only thing that I was thinking when I put it at the end was that we hadn't talked about any rules on Extensions (or the other entities) until that last paragraph, so I wasn't sure if it was bringing up what to do with rules before we talked about what rules could be used.

File: input/pagecontent/reference.md [hunk 2]
```diff
@@ -1273,23 +1275,25 @@ These conformance resources are created using FSH instance grammar. For example,
 
 #### Defining Invariants
 
-Invariants are defined using the declaration `Invariant` and OPTIONAL keywords `Description`<sup>*</sup>, `Severity`<sup>*</sup>, `XPath` (FHIR R4 only) and  `Expression`. The keywords correspond directly to elements in ElementDefinition.constraint. Invariants are incorporated into profiles, extensions, logical models, or resources via [obeys rules](#obeys-rules).
+Invariants are defined using the declaration `Invariant` and OPTIONAL keywords `Description`<sup>1</sup>, `Severity`<sup>2</sup>, `XPath` (FHIR R4 only) and  `Expression`. The keywords correspond directly to elements in ElementDefinition.constraint. Invariants are incorporated into profiles, extensions, logical models, or resources via [obeys rules](#obeys-rules).
```
[the remark this answers is not in the document]
Comment (jafeltra) [reply input/pagecontent/reference.md:1278]: Oh good catch. 

One logistical question - this change wasn't called out in a specific Jira issue. The changes I made here were for [FHIR-42618](https://jira.hl7.org/browse/FHIR-42618) to split out the footnote for `Description` and `Severity`. Should I include this change in the commit to resolve that issue? Or add another commit that just addresses something outside of a ballot issue?

File: input/pagecontent/reference.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jafeltra) [reply input/pagecontent/reference.md]: Fixed up in [4949381](https://github.com/HL7/fhir-shorthand/pull/176/commits/49493816a31842e2023444017b54d4ee26c4454a).

File: input/pagecontent/reference.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jafeltra) [reply input/pagecontent/reference.md]: Added in [d64d505](https://github.com/HL7/fhir-shorthand/pull/176/commits/d64d505f32a8be0b70199399bf357691abe0b94d).

**Review by cmoesel — APPROVED** [review]
Thanks, @jafeltra. This looks great!

**Review by mint-thompson — APPROVED** [review]

**Pull request merged**